nexus/src/app/vpc.rs

```diff
@@ -39,6 +41,42 @@ use uuid::Uuid;
 
 impl super::Nexus {
     // VPCs
+
+    pub fn vpc_lookup<'a>(
+        &'a self,
+        opctx: &'a OpContext,
+        vpc_selector: &'a params::VpcSelector,
+    ) -> LookupResult<lookup::Vpc<'a>> {
+        match vpc_selector {
+            params::VpcSelector {
+                vpc: NameOrId::Id(id),
+                project_selector: None,
+            } => {
+                let vpc =
+                    LookupPath::new(opctx, &self.db_datastore).vpc_id(*id);
+                Ok(vpc)
+            }
+            params::VpcSelector {
+                vpc: NameOrId::Name(name),
+                project_selector: Some(selector),
+            } => {
+                let vpc = self
+                    .project_lookup(opctx, selector)?
+                    .vpc_name(Name::ref_cast(name));
+                Ok(vpc)
+            }
+            params::VpcSelector {
+                vpc: NameOrId::Id(_),
+                project_selector: Some(_),
+            } => Err(Error::invalid_request(
+                "when providing vpc as an ID, project should not be specified",
```

Comment:
`project_selector` itself means `project: ID` or `project: Name, Organization: NameOrId` which is why it's calling out project specifically. It's true that this isn't really complete though, because if `project` was removed but `organization` was left that would probably still be permissible (even if not strictly correct). I'll add a note to create some test cases around these failure modes too.

Reply:
Ah, I see, that helps. I think my confusion began with `instance_selector`, used above. If that were changed to `instance` then these would all be consistent, which seems a marked improvement!

Reply:
I'll add a follow up for that.